File: Java/167.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        while (left < right) {
            int currentSum = numbers[left] + numbers[right];
            if (currentSum == target) {
                return new int[] { left + 1, right + 1 };
            } else if (currentSum > target) {
                right--;
            } else {
                left++;
            }
        }
        return new int[] { -1, -1 };
    }
}

File: Python/167.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1
        cur_sum = numbers[left] + numbers[right]

        while cur_sum != target:
            if cur_sum > target:
                right -= 1
            else:
                left += 1
            cur_sum = numbers[left] + numbers[right]

        return [left + 1, right + 1]
